overlord/configstate/configcore/netplan.go

```diff
@@ -0,0 +1,117 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"strings"
+
+	// TODO: Move to yaml.v3 everywhere, there is PR#10696 that starts
+	//       this. However it is not trivial yaml.v2 accepts duplicated
+	//       keys in maps and v3 does not. There might be snaps in the
+	//       wild that we could break by going to v3.
+	//
+	// Move this part of the code to yaml.v3 because without it we run
+	// into incompatibilites of maps between json and yaml:
+	// "json: unsupported type: map[interface {}]interface {}" because
+	// because yaml.v2 unmarshalls by default to "map[string]interface{}"
```

Comment:
is this right?   or string meant to be interface{} here?

Reply:
Uh, thank you! Good catch, fixed now.